src/main/kotlin/converter/typeUtils.kt

```diff
@@ -115,12 +115,19 @@ private fun ObjectTypeToKotlinTypeMapper.mapUnionType(type: UnionOrIntersectionT
 
     val mappedTypes = notNullTypes.map { mapType(it, null) }
     return KtTypeUnion(when {
-        !nullable -> mappedTypes.distinct()
+        !nullable -> mappedTypes.merge()
         notNullTypes.size == 1 -> mappedTypes.map { it.copy(isNullable = true) }
         else -> (mappedTypes + KtType(NOTHING, isNullable = true)).distinct()
     })
 }
 
+private fun List<KtType>.merge(): List<KtType> {
+    return groupBy { it.copy(comment = null) }.map { entry ->
```

Comment:
`groupBy { it.copy(comment = null) }` looks like a hack to me. We can either define equals explicitly for `KtType` or   or just clearly stat in groupBy call what exactly  we are grouping by.

Reply:
I've commented why it does this.  Making it explicit that comments are not part of the comparison makes it also clear that the function needs to handle concatenating the comments so that they are not lost.

Reply:
Actually, I went ahead and fixed this by making it more explicit.